Makefile

```
@@ -40,7 +40,7 @@ ifeq ($(OSTYPE), darwin)
 	GCOV = xcrun llvm-cov gcov
 endif
 
-BUILDTYPE_LOWER := $(shell echo $(BUILDTYPE) | tr '[A-Z]' '[a-z]')
+BUILDTYPE_LOWER := $(BUILDTYPE) | tr 'A-Z' 'a-z'
```

I don't think that'd work, you need to call `tr` so you need the `shell` function at some point. Also we could use `[:upper:]` and `[:lower:]` char classes in case there are non-ASCII capital letters in the future.

```suggestion
BUILDTYPE_LOWER := $(shell echo $(BUILDTYPE) | tr '[:upper:]' '[:lower:]')
```

---

But doesn't this `Makefile` print `release`?
```Makefile
BUILDTYPE := Release
BUILDTYPE_LOWER := $(BUILDTYPE) | tr 'A-Z' 'a-z'

all:
	@echo $(BUILDTYPE_LOWER)
```
+1 on the char classes. Should I add those in the `grep` commands too?

---

It does, I guess I've never seen this syntax before. Do you know if that's well supported?

> Should I add those in the grep commands too?

+1

---

I'm fairly certain 
```
BUILDTYPE_LOWER := $(BUILDTYPE) | tr 'A-Z' 'a-z'
```

is going to break lines like:
https://github.com/nodejs/node/blob/ddb1cbac476372360e68c7f77470763ce4d2e5a7/Makefile#L280

---

Oh right, try:

```makefile
BUILDTYPE := Release
BUILDTYPE_LOWER := $(BUILDTYPE) | tr 'A-Z' 'a-z'
REAL_BUILDTYPE_LOWER = $(shell echo $(BUILDTYPE) | tr '[:upper:]' '[:lower:]')

all:
	$(info $(BUILDTYPE_LOWER))
	$(info $(REAL_BUILDTYPE_LOWER))
```

You definitely need to call the `shell` in order to store the output of `tr`.